In [1]:
import pandas as pd 
from scipy.stats import pointbiserialr
from scipy.stats import pearsonr
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('general_data.csv')

In [3]:
df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [4]:
df.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [5]:
df.Attrition.replace(to_replace = ('No','Yes'),value = (0,1),inplace = True)

In [6]:
stats, p = pearsonr(df.Attrition, df.Age)
print(stats, p)

-0.15920500686577507 1.996801615893171e-26


# Hypothesis 1

In [7]:
stats, p = pearsonr(df.Attrition, df.DistanceFromHome)
print(stats, p)

-0.009730141010179438 0.5182860428049617


# Hypothesis 2

In [8]:
stats, p = pearsonr(df.Attrition, df.MonthlyIncome)
print(stats, p)

-0.031176281698114025 0.0384274849060192


# Hypothesis 3

In [9]:
stats, p = pearsonr(df.Attrition, df.PercentSalaryHike)
print(stats, p)

0.0325325948910522 0.030743386433369824


# Hypothesis 4

In [16]:
stats, p = pearsonr(df.Attrition, df.TrainingTimesLastYear)
print(stats, p)

-0.049430576244253066 0.001024706191536548


# Hypothesis 5

In [17]:
stats, p = pearsonr(df.Attrition, df.YearsAtCompany)
print(stats, p)


-0.13439221398997386 3.163883122493571e-19


# Hypothesis 6

In [18]:
stats, p = pearsonr(df.Attrition, df.YearsSinceLastPromotion)
print(stats, p)

-0.03301877514258329 0.02833033618939086


# Hypothesis 7

In [19]:

stats, p = pearsonr(df.Attrition, df.YearsWithCurrManager)
print(stats, p)

-0.15619931590162422 1.7339322652951965e-25


# Hypothesis 8

null hypothesis (H0) = Attrition and YearsWithCurrManager have no correlation .

alternate hypothesis (H1) = Attrition and YearsWithCurrManager have a correlation between them .

-0.15619931590162847

1.7339322652896276e-25

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsWithCurrManager.

r is -0.156 so we see there is negative relation between the two